tasklist/els-schema/src/main/java/io/camunda/tasklist/schema/manager/OpenSearchSchemaManager.java

```diff
@@ -64,18 +64,22 @@ public class OpenSearchSchemaManager implements SchemaManager {
   @Override
   public void createSchema() {
     if (tasklistProperties.getArchiver().isIlmEnabled()) {
-      createIndexLifeCycles();
+      createIndexLifeCyclesIfNotExist();
     }
     createDefaults();
     createTemplates();
     createIndices();
   }
 
-  public void createIndexLifeCycles() {
+  public void createIndexLifeCyclesIfNotExist() {
+    if (retryOpenSearchClient.getLifecyclePolicy(TASKLIST_DELETE_ARCHIVED_INDICES).isPresent()) {
+      LOGGER.info("{} ISM policy already exists", TASKLIST_DELETE_ARCHIVED_INDICES);
```

Comment:
This is gonna introduce a not necessary call on Schema Startup, once this is executed first time when it is starting a Tasklist in a new envirovment, and after only when updates happens
I would prefer have this code in a different method, and call on the scope of `ILMPolicyUpdate`, so it would avoid the not need call over here.

Reply:
@nathansandi Actually I have got the failure on Schema Startup, https://github.com/camunda/camunda/pull/18924#discussion_r1619163540

Reply:
> @nathansandi Actually I have got the failure on Schema Startup, [#18924 (comment)](https://github.com/camunda/camunda/pull/18924#discussion_r1619163540)

Yes because probably the schema was created already
So it calls the method from ILM Policy Open Search
but it didnt verify id the policy exist (because the method was updated when return Optional.empty() instead of null)

Reply:
No, I was having the issue before adding this `Optional` staff
The scenario is the following:
- I have started Tasklist 8.5.1
- the ISM policy is created in Opensearch
- then I started Tasklist built from this branch
- it detected there is new index (`process_8.6.0`) so it triggered the createSchema methid
- createSchema calls createIndexLifeCycles (as ILM is enabled) and gets 409 from Opensearch because the policy already exists
- server fails to start

am I missing something?

Reply:
Yes, the line 54 that was deleted on ILMOpenSearch was doing this verification

It is not necessary to do this verification on [OpenSearchSchemaManager.java](https://github.com/camunda/camunda/pull/18924/files/5c66bf7b456247beb39ab2f3d50d5a21691ffdd2#diff-505521aedf4c21037703b1d0304bada59e17de0095b4629c38a05529874f7924) 

If you delete the images of open search and create again, you will be add an unnecessary GET, once it will create the policy before the Schema (The code is called when the schema does not exist) 

After we can check if already exists only on the scope of ILMOpenSearch 
This line was removed here: https://github.com/camunda/camunda/pull/18924/commits/e0c3a74b6246e2891a3322554edb6f87e693d331#diff-e933dcceda7da04d7a8cefb97ea7ee1f8c013a8e6059e34a183c122d7aa67849L51 

Was just update to empty instead of null .

Reply:
`schemaExists()` returns true when there is at least one missing index or the number of replicas is not as configured, so it means the ISM policy it can be already present at that moment. https://github.com/camunda/camunda/blob/9d90ffdcca14f3c3d8a6ca3c2bda8e758e395707/tasklist/els-schema/src/main/java/io/camunda/tasklist/os/IndexSchemaValidatorOpenSearch.java#L133-L145